teradata/datatypes.py

```diff
@@ -230,7 +230,9 @@ def convertValue(self, dbType, dataType, typeCode, value):
         logger.trace(
             "Converting \"%s\" to (%s, %s).", value, dataType, typeCode)
         if value is not None:
-            if typeCode == NUMBER:
+            if typeCode == STRING:
```

Comment:
Running the unit tests, this actually breaks conversion for JSON, INTERVAL, and PERIOD data types as these types don't have an equivalent in Python Database API specification and thus were mapped to STRING typecode.   

If we want to short circuit String, then I think we have to introduce new type codes for JSON, INTERVAL, and PERIOD.   Let me know if you want to submit or new pull request or otherwise I can make the change.

Reply:
Ah, that makes sense. I updated this PR to include new type codes for these three types. Hopefully this helps fix the tests.

Caveat: I'm not super familiar with the Python DB API specs.